scholia/app/templates/chemical_related.sparql

```diff
@@ -1,12 +1,29 @@
 PREFIX target: <http://www.wikidata.org/entity/{{ q }}>
 
 # title: related chemical structures
-SELECT ?mol ?molLabel ?InChIKey ?CAS ?ChemSpider ?PubChem_CID WHERE {
-  target: wdt:P235 ?queryKey .
-  ?mol wdt:P235 ?InChIKey .
+SELECT ?mol ?molLabel ?InChIKey ?CAS ?ChemSpider ?PubChem_CID WITH {
+  SELECT ?queryKey ?srsearch ?filter WHERE {
+    target: wdt:P235 ?queryKey .
+    BIND (CONCAT(substr($queryKey,1,14), " haswbstatement:P235") AS ?srsearch)
+    BIND (CONCAT("^", substr($queryKey,1,14)) AS ?filter)
+  }
+} AS %MOLS WITH {
+  SELECT ?mol ?InChIKey WHERE {
+    INCLUDE %MOLS
+    SERVICE wikibase:mwapi {
+        bd:serviceParam wikibase:endpoint "www.wikidata.org";
+        wikibase:api "Search";
+        mwapi:srsearch ?srsearch;
+        mwapi:srlimit "max".
+        ?mol wikibase:apiOutputItem mwapi:title.
+      }
+    ?mol wdt:P235 ?InChIKey .
+    FILTER (regex(str(?InChIKey), ?filter))
```

Comment:
I usually also add
`FILTER (?InChIKey != ?queryKey)`
at line 22

Reply:
Ah, thanks for the reminder. I originally had that, but the query was already so slow, I left this out, making it at least somewhat faster. But now I can put this in!